### ISSUE TITLE
Make PiwikLdap officially supported by Piwik core team?

### ISSUE THREAD

**Issue by mattab** [piwik/plugin-LoginLdap#33]
Dear @tehnotronic 

Thank you for this great plugin! the PiwikLdap plugin is we think a very important feature for Piwik. We would like to make the plugin an official Piwik plugin. 
Some of the things we would like to do:
- add new features
- maintain the plugin, so that it never breaks when we change the core APIs (we'll add it as a [Submodule](https://github.com/piwik/piwik/blob/master/.gitmodules))
- fix bugs when they are reported

To make this plugin official, we would need to:
- move it to https://github.com/piwik/plugin-LoginLdap
- in the plugin.json we would replace `Matthieu Aubry` by `Piwik`

Of course you would still have push access to the repository and ability to close/open tickets. It's just that by moving the plugin to piwik/ organisation, we would now officially support it and make it even better :)

FYI here are some of the features we would like to add in LoginLdap plugin:
- create user interface to customise some Ldap settings
- let user add several Ldap servers (and order them) so that we can fallback in case the ldap  master is not available
  - each Ldap server is defined by: server url, port, username, optional password, base DN)
- let user know whether the connection to Ldap server(s) work
- let user customise which Ldap attribute contain the fields: 
  - username / UID
  - first name
  - last name 
  - note: first name and last name will be concatenated to make the Alias
  - email address  
  - (maybe the list of available attributes could be displayed to user for her to pick values from)
- let user simply customise the raw `filter` attribute value string
  - for example user may set it to: `(&(resMemberOf=cn=piwikMTO,ou=commonOU,dc=server,dc=mydcname_here,dc=fr)(objectclass=person))` which will authorize Ldap objects of type `person` and that have an ‘resMemberOf’ attribute value of: 'cn=piwikMTO,ou=commonOU,dc=serveur,dc=mydnname_here,dc=fr’
- let user specify a custom Ldap attribute name, which is a multivariate attribute that contains the permission level for this user, eg:
  - `view:all` gives the user view access to all websites, and `admin:all` gives user admin access to all websites.
  - `superuser` gives the user Super User permission
  - `view:4` or `admin:4` gives specified permission to website id 4
  - these permissions will be synchronised regularly (every hour?) to ensure the permission level set in Ldap is up to date in Piwik
- let user 
- create automated tests 
- ensure compatibility with OpenDirectory (LDAP v3)

in summary we want to make the plugin a bit easier to use, and implement some of the ideas from the [OwnCloud Ldap integration](http://doc.owncloud.org/server/6.0/admin_manual/configuration/auth_ldap.html)

**What do you think?**
